### PR TITLE
Add credentials secret to AWS provider config in order to support CredentialsRequest

### PR DESCRIPTION
**Pull request by spangenberg**
/cc bison

### CODE REVIEW

**Comment by dgoodwin** [comment]
If we do this by setting a secret on every MachineSet, and ultimately remove the IAM role that allows the machine API to work today, the default state for anyone who creates a MachineSet would be broken?

Just wondering if this would be best done by default in the machine-api, rather than having it pinned on every  MachineSet.

**Comment by bison** [comment]
> If we do this by setting a secret on every MachineSet, and ultimately remove the IAM role that allows the machine API to work today, the default state for anyone who creates a MachineSet would be broken?
> 
> Just wondering if this would be best done by default in the machine-api, rather than having it pinned on every MachineSet.
> 

Any defaults would have to go into the controllers in our fork of the cluster-api, and further deviate from upstream. Users already have to choose quite a few things in the `providerConfig` that we can't provide sensible defaults for. Either that, or they copy an existing `MachineSet` created by the installer. The latter is probably the more common case right now.

We probably want to start using `MachineClass` objects for this kind of thing, but I'm not sure where we're at with that.

**Comment by dgoodwin** [comment]
I understand, thanks!

**Comment by bison** [comment]
This isn't going to pass until https://github.com/openshift/machine-api-operator/pull/199 is merged.

**Comment by spangenberg** [comment]
> This isn't going to pass until [openshift/machine-api-operator#199](https://github.com/openshift/machine-api-operator/pull/199) is merged.

@bison you sure, because they still have the same thing in their repository:
https://github.com/openshift/cloud-credential-operator/blob/47261953fd5974ba24072148d53ec4c919f6ad8f/manifests/0000_30_cloud-credential-operator_03_cred-openshift-cluster-api.yaml

**Comment by bison** [comment]
@spangenberg, ah, I forgot about that. That is using the wrong namespace though. We changed it to `openshift-machine-api`.

**Comment by spangenberg** [comment]
/test e2e-aws

https://github.com/openshift/machine-api-operator/pull/199 got merged 🎉

**Comment by enxebre** [comment]
can we then get rid of https://github.com/openshift/installer/blob/master/data/data/aws/master/main.tf#L36 as part of this PR? cc @dgoodwin @abhinavdahiya

**Comment by dgoodwin** [comment]
Hopefully we'd be at a point where we can try at least, I think that was an end goal of all this.

**Comment by spangenberg** [comment]
Machine controller is throwing errors:
```E0228 16:31:15.171928       1 instances.go:310] error creating EC2 instance: UnauthorizedOperation: You are not authorized to perform this operation. Encoded authorization failure message: qM5RU2SzLuICZYAmY10kX86MSaRNCbibh7MjtAO6tV0oBYF2oEigg3sv4jVvck9l8_Lq40Jd9nvvweD_XTY6EAZvZ94yNidNVDOQcChaOewOZWXSZWHsVIKRWckGMrDYRu84TxnxsV2EdbHqHQVhL39GFuyBgFUhsx2i0hj2Ep2qYmUk2_JU4tTlHN7_iS-hAY5in0qx5gEHKrZw16dNpiHzsvEg1o6HN4krQDoSqPae8FswngSd-1PaRPlIXCqBFO28D34i5BfrYH7-Sc-6K8C99hcjDullKTH_Wb8EDU5VgPIKdFHDOP5tC1XdmNvCYLAlRHdQYJaTT3GCHymx9RD3sIFK5HLNwcGBAeCZ--9qFGhtnJWx91nGD1pps8DqAONLvmOUXNc6jLqdPUZm_OGSVgX5VpgOpa5onzQa-jAxdRK1iNBZtYWWBvIAwfo-T1fjSWDlK5BpRRs1aZEAOIg63rD37ey3F7eMLMqYMrygyBKpXEo8mTTqMdm47xOFzmoMfq6oD0tP4OKEJtThwS1NfJd6Nt-6qJ9f6wVTo-TdxujlYChx-haxc5L2PMh1Y6D_5itSYy_5GiU3GcJfUUq1AqJXUagyExVJCyf8LWN8b-vSUJOFm3fwbuDLDqJWPvvV5NNFOERZBZ9yiPF_m2zKqmLQn--bNa__SqzqIIGgFE1Ifouw-tKZqaNgM3Nbzc8zOx7O8oNPoqpCYVgeMwtsyyS0iAWPFdtjviBlFOQgaMFoB7B3z_M_UFA
	status code: 403, request id: 6a61f19a-5257-4a07-bd58-35eb654c7a7d
E0228 16:31:15.172013       1 actuator.go:106] Machine error: error launching instance: error creating EC2 instance: UnauthorizedOperation: You are not authorized to perform this operation. Encoded authorization failure message: qM5RU2SzLuICZYAmY10kX86MSaRNCbibh7MjtAO6tV0oBYF2oEigg3sv4jVvck9l8_Lq40Jd9nvvweD_XTY6EAZvZ94yNidNVDOQcChaOewOZWXSZWHsVIKRWckGMrDYRu84TxnxsV2EdbHqHQVhL39GFuyBgFUhsx2i0hj2Ep2qYmUk2_JU4tTlHN7_iS-hAY5in0qx5gEHKrZw16dNpiHzsvEg1o6HN4krQDoSqPae8FswngSd-1PaRPlIXCqBFO28D34i5BfrYH7-Sc-6K8C99hcjDullKTH_Wb8EDU5VgPIKdFHDOP5tC1XdmNvCYLAlRHdQYJaTT3GCHymx9RD3sIFK5HLNwcGBAeCZ--9qFGhtnJWx91nGD1pps8DqAONLvmOUXNc6jLqdPUZm_OGSVgX5VpgOpa5onzQa-jAxdRK1iNBZtYWWBvIAwfo-T1fjSWDlK5BpRRs1aZEAOIg63rD37ey3F7eMLMqYMrygyBKpXEo8mTTqMdm47xOFzmoMfq6oD0tP4OKEJtThwS1NfJd6Nt-6qJ9f6wVTo-TdxujlYChx-haxc5L2PMh1Y6D_5itSYy_5GiU3GcJfUUq1AqJXUagyExVJCyf8LWN8b-vSUJOFm3fwbuDLDqJWPvvV5NNFOERZBZ9yiPF_m2zKqmLQn--bNa__SqzqIIGgFE1Ifouw-tKZqaNgM3Nbzc8zOx7O8oNPoqpCYVgeMwtsyyS0iAWPFdtjviBlFOQgaMFoB7B3z_M_UFA
	status code: 403, request id: 6a61f19a-5257-4a07-bd58-35eb654c7a7d
E0228 16:31:15.172025       1 actuator.go:115] error creating machine: error launching instance: error creating EC2 instance: UnauthorizedOperation: You are not authorized to perform this operation. Encoded authorization failure message: qM5RU2SzLuICZYAmY10kX86MSaRNCbibh7MjtAO6tV0oBYF2oEigg3sv4jVvck9l8_Lq40Jd9nvvweD_XTY6EAZvZ94yNidNVDOQcChaOewOZWXSZWHsVIKRWckGMrDYRu84TxnxsV2EdbHqHQVhL39GFuyBgFUhsx2i0hj2Ep2qYmUk2_JU4tTlHN7_iS-hAY5in0qx5gEHKrZw16dNpiHzsvEg1o6HN4krQDoSqPae8FswngSd-1PaRPlIXCqBFO28D34i5BfrYH7-Sc-6K8C99hcjDullKTH_Wb8EDU5VgPIKdFHDOP5tC1XdmNvCYLAlRHdQYJaTT3GCHymx9RD3sIFK5HLNwcGBAeCZ--9qFGhtnJWx91nGD1pps8DqAONLvmOUXNc6jLqdPUZm_OGSVgX5VpgOpa5onzQa-jAxdRK1iNBZtYWWBvIAwfo-T1fjSWDlK5BpRRs1aZEAOIg63rD37ey3F7eMLMqYMrygyBKpXEo8mTTqMdm47xOFzmoMfq6oD0tP4OKEJtThwS1NfJd6Nt-6qJ9f6wVTo-TdxujlYChx-haxc5L2PMh1Y6D_5itSYy_5GiU3GcJfUUq1AqJXUagyExVJCyf8LWN8b-vSUJOFm3fwbuDLDqJWPvvV5NNFOERZBZ9yiPF_m2zKqmLQn--bNa__SqzqIIGgFE1Ifouw-tKZqaNgM3Nbzc8zOx7O8oNPoqpCYVgeMwtsyyS0iAWPFdtjviBlFOQgaMFoB7B3z_M_UFA
	status code: 403, request id: 6a61f19a-5257-4a07-bd58-35eb654c7a7d
```

**Comment by enxebre** [comment]
some perm must me missed, you should be able to use `you should be able to use `aws sts decode-authorization-message --encoded-message` and go from there

**Comment by spangenberg** [comment]
This should be unblocked by https://github.com/openshift/cloud-credential-operator/pull/37 and additionally https://github.com/openshift/machine-api-operator/pull/241.

**Comment by spangenberg** [comment]
Depending PRs are all merged now.

/retest

File: data/data/aws/master/main.tf [hunk 1]
```diff
@@ -33,41 +29,6 @@ EOF
   ), var.tags)}"
 }
 
-resource "aws_iam_role_policy" "master_policy" {
```
Comment (abhinavdahiya) [inline data/data/aws/master/main.tf]: We need the iam policy for kubelet to running. also the kube-controller-manager would need access to AWS api too.

Comment (spangenberg) [reply data/data/aws/master/main.tf]: I removed the change, something like this can happen in another PR if wanted.

**Pull request merged**